src/project-css/app/Http/Controllers/WizardController.php

```diff
@@ -0,0 +1,294 @@
+<?php
+/**
+ * @author Tracy A McCormick <tam0013@mail.wvu.edu>
+ */
+
+namespace App\Http\Controllers;
+
+use App\Models\Collection;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Storage;
+use App\Helpers\TableHelper;
+use Auth;
+
+class WizardController extends Controller
+{
+    private $strDir;
+    private $sharedViewData;
+
+    public function __construct($storageFolder = 'flatfiles') {
+      // Protection to make sure this is only accessible to admin
+      $this->middleware('admin');
+
+      // Storage directory
+      $this->strDir = $storageFolder;
+
+      // Shared View Data is used by both the flatfile and cms
+      // functions to correctly render the file import views.
+      $this->sharedViewData = [
+            'AuthUsr' => Auth::user(),
+            'collcntNms' => Collection::all(),
+            'fltFleList' => $this->flatFileList(),
+            'colID' => 1
+       ];
+    } 
+
+   /**
+    * Returns view for the Import Wizard Main Page
+    *
+    * @return view
+    */    
+    public function import() {
+        return view('admin/wizard/import')->with('AuthUsr', Auth::user());
+    }
+
+    public function importCollection($colID) {
+        // get shared view data
+        $data = $this->sharedViewData;
+        
+        // set collection id to current
+        $data['colID'] = $colID;
+
+        if (Collection::findorFail($colID)->isCms) {
+            // return cms import wizard
+            return view('admin/wizard/cms')->with($data);
+        }
+
+        // return flatfile import wizard
+        return view('admin/wizard/flatfile')->with($data);
+    }
+
+   /**
+    * Returns view for the CMS Import
+    *
+    * @return view
+    */      
+    public function cms() {
+        return view('admin/wizard/cms')->with($this->sharedViewData);
+    }
+
+   /**
+    * Returns view for the FlatFile Import
+    *
+    * @return view
+    */      
+    public function flatfile() {
+        return view('admin/wizard/flatfile')->with($this->sharedViewData);
+    }
+
+   /**
+    * Process form request to create a new table in a collection
+    *
+    * Prepares an array of values required by storeUploadsAndImport
+    * to continue processing the import request.
+    *
+    * @param request The request object has 2 fields that we require for the import
+    * flatFiles should be an array of files and the imprtTblNme (New Table Name)
+    *
+    * @return redirect
+    */       
+    public function flatfileUpload(Request $request) {
+        // 1. Input the table name in the meta Directory
+        //Rules for validation
+        $rules = array(
+            'imprtTblNme' => 'required|unique:tables,tblNme|max:30|min:6|alpha_num',
+            'colID' => 'required|Integer',
+            'fltFile' => 'required|file|mimetypes:text/plain|mimes:txt,dat,csv,tab',
+        );
+
+        //Customize the error messages
+        $messages = array(
```

Comment:
I like this solution type for removing large amounts of conditionals, should probably move it to a helper function or define in configs similar to I18n in ruby.